notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/main/kotlin/com/r3/corda/notary/plugin/nonvalidating/server/NonValidatingNotaryServerFlowImpl.kt

```diff
@@ -132,13 +131,13 @@ class NonValidatingNotaryServerFlowImpl() : ResponderFlow {
                 transactionSignatureService.signBatch(listOf(txDetails), listOf(requestPayload.notaryKey)).first().first()
             } else null
 
-            session.send(uniquenessResult.toNotarisationResponse(signature))
+            session.send(uniquenessResult.toNotarisationResponse(txDetails.id, signature))
         } catch (e: Exception) {
             logger.warn("Error while processing request from client. Cause: $e ${e.stackTraceToString()}")
             session.send(
                 NotarisationResponse(
                     emptyList(),
-                    NotaryErrorGeneralImpl("Error while processing request from client.", e)
+                    NotaryExceptionGeneral("Error while processing request from client.")
```

Comment:
```suggestion
                    NotaryExceptionGeneral("Error while processing request from client. Please contact notary operator for further details.")
```

Optional suggestion - now we've (intentionally) taken away the cause details, it might be worth making it clear how the caller can find out more. The notary operator should be able to refer to their logs.